src/clients/MultiCallBundler.ts

```diff
@@ -17,18 +25,97 @@ export class MultiCallBundler {
     this.transactions = [];
   }
 
+  getTarget(index: number) {
+    return { target: this.transactions[index].contract.address };
+  }
+
   async executeTransactionQueue() {
-    // TODO: this should include grouping logic for multicall
-    this.logger.debug({ at: "MultiCallBundler", message: "Executing tx bundle", number: this.transactions.length });
-    const transactionResults = await Promise.all(this.transactions);
-    const transactionReceipts = await Promise.all(transactionResults.map((transaction: any) => transaction.wait()));
-
-    // TODO: add additional logging on error processing.
-    this.logger.debug({
-      at: "MultiCallBundler",
-      message: "All transactions executed",
-      hashes: transactionReceipts.map((receipt) => receipt.transactionHash),
-    });
-    return transactionReceipts;
+    try {
+      this.logger.debug({ at: "MultiCallBundler", message: "Executing tx bundle", number: this.transactions.length });
+
+      // Simulate the transaction execution for the whole queue.
+      const transactionsSucceed = await Promise.all(
+        this.transactions.map((tx: preSignedTransaction) => willSucceed(tx.contract, tx.method, tx.args))
+      );
+
+      // If any transactions will revert then log the reason and remove them from the transaction queue.
+      if (transactionsSucceed.some((willSucceed) => !willSucceed.succeed))
```

Comment:
```suggestion
      if (transactionsSucceed.some(({ succeed }) => !succeed))
```

Just because I think it's a little confusing shadowing the function name with the variable name.